src/watchdog/observers/api.py

```diff
@@ -377,3 +378,8 @@ def dispatch_events(self, event_queue):
                 if handler in self._handlers.get(watch, []):
                     handler.dispatch(event)
         event_queue.task_done()
+
+
+class BaseObserverSubclassCallable(Protocol):
```

Comment:
Nice!

Reply:
I dunno, I think I might rather rework the inheritance so that the init calls are "properly" compatible.  But, this is a game to play for the moment.

Reply:
The "nice" was mostly for the `Protocol` usage :)
I never really used that pattern, and it may be quite helpful!

Reply:
Definitely, I tend away from inheritance and `Protocol` certainly gives you a nice static way to describe shared interfaces.